lib/neo4j/core/cypher_session/adaptors/faraday_helpers.rb

```diff
@@ -0,0 +1,49 @@
+module Neo4j
+  module Core
+    class CypherSession
+      module Adaptors
+        module FaradayHelpers
+          private
+
+          def verify_faraday_options(faraday_options = {}, defaults = {})
+            faraday_options.symbolize_keys!.reverse_merge!(defaults)
+            faraday_options[:adapter] ||= :net_http_persistent
+            require 'typhoeus/adapters/faraday' if faraday_options[:adapter].to_sym == :typhoeus
```

Comment:
Shouldn't the user `require` the gem paths that they need depending on the adaptor that they are using?  That way we wouldn't need to include the various gems in our `Gemfile`